src/interop/pointer.jl

```diff
@@ -114,6 +114,31 @@ Base.:(+)(x::Integer, y::LLVMPtr) = y + x
 Base.unsigned(x::LLVMPtr) = UInt(x)
 Base.signed(x::LLVMPtr) = Int(x)
 
+@generated function Base.reinterpret(::Type{LLVMPtr{TDest, ASDest}}, src::LLVMPtr{TSrc, ASSrc}) where {TDest, ASDest, TSrc, ASSrc}
+    @dispose ctx=Context() begin
+        T_dest = convert(LLVMType, LLVMPtr{TDest, ASDest}; ctx)
+        T_src = convert(LLVMType, LLVMPtr{TSrc, ASSrc}; ctx)
+
+        llvm_f, _ = create_function(T_dest, [T_src])
+        mod = LLVM.parent(llvm_f)
+
+        @dispose builder=Builder(ctx) begin
+            entry = BasicBlock(llvm_f, "entry"; ctx)
+            position!(builder, entry)
+
+            dest_ptr = if ASDest != ASSrc
```

Comment:
If `TDest` and `TSrc` are different you need additionally a bitcast

Reply:
AFAICT for `LLVMPtr` the compiler ignores the pointer eltype and always use `i8` instead, so I'm not sure how to actually get the `bitcast` to show up.